jax/_src/nn/initializers.py

```diff
@@ -44,6 +44,13 @@ def init(key, shape, dtype=dtype):
     return random.normal(key, shape, dtype) * stddev
   return init
 
+def truncated_normal(stddev=1e-2, lower=-2, upper=2, dtype=jnp.float_):
```

Comment:
I think you can factor https://github.com/google/jax/pull/8346/files#diff-dfe359c36c6d226fc1df3447c75b7d39f409e4d145221a2d7597147f45c51b1fR126-R133 into a helper and then call it here? That way this will also work for complex dtypes.